src/tree/transform.js

```diff
@@ -79,33 +80,116 @@ export const applyTransform = (eventTransform, svg, divBox, width, height) => {
   );
 };
 
-export function setZooming({ containerData, setTransform, transformState, wrapperState, initialZoomState }) {
-  const { svg, divBox, width, height, zoomWrapper, element, tooltip } = containerData;
+export function setZooming({
+  containerData,
+  setTransform,
+  transformState,
+  wrapperState,
+  initialZoomState,
+  interactions,
+}) {
+  const { svg, divBox, width, height, zoomWrapper, tooltip } = containerData;
   const { x = 0, y = 0 } = transformState;
   const { minZoom, maxZoom } = constants;
   const zoomFactor = (transformState && 1 / transformState.zoom) || initialZoomState.initialZoom;
   const scaleFactor = Math.max(Math.min(maxZoom, zoomFactor), minZoom);
 
-  // sends otherwise captured mouse event to handle context menu correctly in sense
-  const bubbleEvent = () => {
-    const newEvent = document.createEvent('MouseEvents');
-    newEvent.initEvent('mousedown', true, false);
-    element.dispatchEvent(newEvent);
+  const translate = (e, data, saveState) => {
+    if (!wrapperState.constraints.active) {
+      closeTooltip(tooltip);
+      const { deltaX, deltaY } = data.swipe;
+      const newX = wrapperState.transform.x + deltaX;
+      const newY = wrapperState.transform.y + deltaY;
+      applyTransform({ x: newX, y: newY, zoom: wrapperState.transform.zoom }, svg, divBox, width, height);
+      if (saveState) {
+        setTransform({ x: newX, y: newY, zoom: wrapperState.transform.zoom });
+        setTimeout(() => {
```

Comment:
why is the timeout needed?

Reply:
When you stop swiping or zooming there is a touch tap event triggered (the end event). We don't want to trigger the selection state at the end of zooming/panning. If you remove the timeout you the swiping state would be false again when you reach the touch tap end event

Reply:
Need to keep an eye on this. There should be a better way, but I know it is generally very problematic at times. Could you move the timeout and swiping = false to the end() handler instead? just to make things clearer? keeping it in translate kind of hide it

Reply:
Yeah that should be no problem